port_ocean/core/handlers/entity_processor/jq_entity_processor.py

```diff
@@ -68,15 +74,19 @@ async def _search_as_object(
 
         return result
 
-    async def _get_entity_if_passed_selector(
+    async def _get_mapped_entity(
         self,
         data: dict[str, Any],
         raw_entity_mappings: dict[str, Any],
         selector_query: str,
+        ignore_selector: bool = False,
     ) -> dict[str, Any]:
         should_run = await self._search_as_bool(data, selector_query)
-        if should_run:
-            return await self._search_as_object(data, raw_entity_mappings)
+        if ignore_selector or should_run:
+            mapped_entity = await self._search_as_object(data, raw_entity_mappings)
+            mapped_entity[DOES_ENTITY_PASSED_SELECTOR_KEY] = should_run
+            return mapped_entity
```

Comment:
I am not a fan at all of passing flags through unrelated objects, you can change the data structure of how you return that data, we can discuss it offline, but injecting into an object which is user related shouldn't be done, and is very hard to debug.